// FILE: a.kt
@file:JvmName("Util")
@file:JvmMultifileClass

<!CONFLICTING_JVM_DECLARATIONS!>fun main(args: Array<String>)<!> {}

// FILE: b.kt
@file:JvmName("Util")
@file:JvmMultifileClass

<!CONFLICTING_JVM_DECLARATIONS!>fun main(args: Array<String>)<!> {}